File: src/main/java/ma/fstg/gestionprojets/services/OrganismeService.java
package ma.fstg.gestionprojets.services;
import ma.fstg.gestionprojets.dto.request.OrganismeRequest;
import ma.fstg.gestionprojets.dto.response.OrganismeResponse;
import java.util.List;
public interface OrganismeService {
    OrganismeResponse create(OrganismeRequest req);
    OrganismeResponse update(Long id, OrganismeRequest req);
    OrganismeResponse getById(Long id);
    List<OrganismeResponse> getAll();
    List<OrganismeResponse> searchByNom(String nom);
    void delete(Long id);
}

File: gestion-projets-frontend/src/services/organismeService.ts
import api from '../utils/axiosConfig';

export const organismeService = {
  getAll: () => api.get('/organismes'),
  getById: (id: number) => api.get(`/organismes/${id}`),
  create: (data: any) => api.post('/organismes', data),
  update: (id: number, data: any) => api.put(`/organismes/${id}`, data),
  delete: (id: number) => api.delete(`/organismes/${id}`),
};
